src/sentry/db/deletion.py

```diff
@@ -127,3 +110,100 @@ def execute(self, chunk_size=10000):
             self.execute_postgres(chunk_size)
         else:
             self.execute_generic(chunk_size)
+
+    def iterator(self, chunk_size=100):
+        if db.is_postgres():
+            g = self.iterator_postgres(chunk_size)
+        else:
+            g = self.iterator_generic(chunk_size)
+
+        for chunk in g:
+            yield chunk
+
+    def iterator_postgres(self, chunk_size, batch_size=1000000):
```

Comment:
Some notes:

- 1000000 was chosen totally arbitrarily
- This method now acts a lot like the `continuous_*` methods above and might make sense to change the name to reflect that, as well as updating the generic iterator to follow the same pattern
- This can infinite loop if the caller doesn't actually delete the returned rows for some reason (e.g. there's an exception that prevents the delete from occurring and it's caught and ignored). This can also return the same row as one that's waiting in the queue to be deleted from the last iteration(s) but as long as it's eventually deleted, the loop will exit